Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -417,8 +417,41 @@ private string GetFunctionLogPath(string name)
                 }
             }
 
-            return await FileSystemHelpers.ReadAllTextFromFileAsync(testDataFilePath);
+            // TODO use Rabin-Karp
+            var fileContentString = await FileSystemHelpers.ReadAllTextFromFileAsync(testDataFilePath);
+            // if it contains the following json property, need to remove them from the file
+            // "code": {
+            //    "name": "code",
+            //    "value": "gibberishxyz..."
+            // }
+            JObject testDataToAdjust = null;
+            if (fileContentString.Contains("\"code\":"))
+            {
+                try
+                {
+                    var testData = JObject.Parse(fileContentString);
+                    var codeProperty = testData["code"];
+                    if ("code".Equals((string)codeProperty["name"], StringComparison.OrdinalIgnoreCase) && (string)codeProperty["value"] != null)
+                    {
+                        testDataToAdjust = testData;
+                    }
+                }
+                catch (Exception)
+                {
+                    // failed to retrieve code property from fileContentString, NOOP
+                }
+            }
+
+            if (testDataToAdjust != null)
+            {
+                // TODO logging, since we are touching user data
+                testDataToAdjust.Property("code").Remove();
```

Comment:
Any reason to not do that directly under the `if ("code".Equals(...))` and avoid the extra `testDataToAdjust`?

Reply:
```
testDataToAdjust.Property("code").Remove();
// update the fileContentString
fileContentString = testDataToAdjust.ToString(Formatting.None);
await FileSystemHelpers.WriteAllTextToFileAsync(testDataFilePath, fileContentString);
```
Are you talking about putting this whole chunk in here?
```
if ("code".Equals((string)codeProperty["name"], StringComparison.OrdinalIgnoreCase) && (string)codeProperty["value"] != null)
{
	testDataToAdjust = testData;
}
```
I don't want to do that because I am ignoring any failures from deserialization and parsing, they simply implies that testData needs no further adjustments
```
catch (Exception)
{
	// failed to retrieve code property from fileContentString, NOOP
}
```
And I don't want to swallow any exceptions after the format is matched, ie IO exceptions during rewrite

Reply:
Ok fair enough.